### PR TITLE
Added Twitter Search

### LINKED ISSUE #420: Add Twitter search
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behavior:**
There's no feature to search tweets.

**Expected behavior:**
There should be a feature to search tweets

I would like to work on this 😄

### PR DESCRIPTION
**Pull request by Remorax**
Fixes #420 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Added Twitter search by making a request to the Loklak API

@ParthS007 @vaibhavsingh97  Please review :)

  
screen shot 2018-01-06 at 10 21 03 pm
screen shot 2018-01-06 at 10 21 23 pm

### CODE REVIEW

File: app/scrapers/__init__.py [hunk 1]
```diff
@@ -44,6 +46,8 @@ def feed_gen(query, engine, count=10):
     engine = old_names.get(engine, engine)
     if engine in ('quora', 'youtube'):
         urls = scrapers[engine].search_without_count(query)
+    elif engine == 'twitter':
```
Comment (cclauss) [inline app/scrapers/__init__.py]: This change is not needed.

File: app/scrapers/twitter.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    def search(self, query, count):
+        """ Makes a GET request to Loklak API and returns the URLs
+        Returns: urls (list)
+                [[Title1,url1], [Title2, url2],..]
+        """
+        url = self.loklakURL+query
+
+        bytesString = urllib.request.urlopen(url).read()
+        dictString = bytesString.decode('utf-8')
+        responses = ast.literal_eval(dictString)
```
Comment (cclauss) [inline app/scrapers/twitter.py]: responses = requests.get(url).json()

File: app/scrapers/twitter.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import urllib.request
```
Comment (cclauss) [inline app/scrapers/twitter.py]: The other scrapers are using Requests https://github.com/fossasia/query-server/blob/master/app/scrapers/generalized.py#L2  Doing so will save you a ton of aggravation.

**Comment by Remorax** [comment]
I've made the necessary changes and squashed the commits also. Please check @cclauss 😄

File: app/scrapers/twitter.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import requests
+
+
+class Twitter():
+    """docstring for Twitter"""
+
+    def __init__(self):
+        self.loklakURL = 'http://api.loklak.org/api/search.json?\
+                            timezoneOffset=-330&q='
```
Comment (cclauss) [inline app/scrapers/twitter.py]: PEP8 advises that we avoid backslashes like the plague that they are.  The problem is that one whitespace to the right of the backslash and the code stops working but the change is invisible to the reader.  The recommended approach is to use parenthesis: 

```python
self.loklakURL = ('http://api.loklak.org/api/search.json?'
                  'timezoneOffset=-330&q=')

File: app/scrapers/twitter.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import requests
+
+
+class Twitter():
+    """docstring for Twitter"""
```
Comment (cclauss) [inline app/scrapers/twitter.py]: Something a bit more explanatory here?  "This scraper takes a query and a count and returns the results of a Twitter search which is executed via the Loklak API."

File: app/scrapers/twitter.py (hunk 4 above)
Comment (cclauss) [reply app/scrapers/twitter.py]: Also is timezoneOffset=-330 really essential?

**Comment by cclauss** [comment]
The search engine needs to be added at https://github.com/fossasia/query-server/blob/master/app/test_server.py#L33-L34 to that tests are done on it.

File: app/scrapers/twitter.py (hunk 4 above)
Comment (ParthS007) [reply app/scrapers/twitter.py]: I don't think it's essential @cclauss, Can you please remove this @Remorax.

File: app/scrapers/twitter.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        Returns: urls (list)
+                [[Title1,url1], [Title2, url2],..]
+        """
+        url = self.loklakURL+query
+
+        responses = requests.get(url).json()
+
+        tweets = []
+        for response in responses['statuses']:
+            tweets.append({'link': response['link'], 'tweet': response['text']})
```
Comment (cclauss) [inline app/scrapers/twitter.py]: Does downstream expect the dict keys to be ('title', 'link') like [all other scrapers](https://github.com/fossasia/query-server/blob/master/app/scrapers/google.py#L26)?

Comment (cclauss) [reply app/scrapers/twitter.py]: Do we want to print the results like the other scrapers do?

File: app/scrapers/twitter.py (hunk 5 above)
Comment (Remorax) [reply app/scrapers/twitter.py]: I have added an explanation near the Search function which amounts to the same thing, but yeah sure I'll include this as well 😄

File: app/scrapers/twitter.py (hunk 4 above)
Comment (Remorax) [reply app/scrapers/twitter.py]: Okay

File: app/scrapers/twitter.py (hunk 6 above)
Comment (Remorax) [reply app/scrapers/twitter.py]: Should I change it back to link and title?

Comment (cclauss) [reply app/scrapers/twitter.py]: We need @ParthS007 to weigh in on that question.  Perhaps it is OK as is.

**Comment by vaibhavsingh97** [comment]
Please resolve conflicts. Thanks

**Comment by Remorax** [comment]
Closing this PR. Will open another one soon

**Pull request closed without merging**